ecs-agent/wsclient/client.go

```diff
@@ -565,3 +589,42 @@ func permissibleCloseCode(err error) bool {
 		websocket.CloseGoingAway,         // websocket error code 1001
 		websocket.CloseInternalServerErr) // websocket error code 1011
 }
+
+// newDisconnectTimeoutHandler returns new timer object to disconnect from server connection start time, with goroutine
+// to disconnect from client.
+func (cs *ClientServerImpl) newDisconnectTimeoutHandler(startTime time.Time,
+	metricName string,
+	disconnectTimeout time.Duration,
+	disconnectJitterMax time.Duration) *time.Timer {
+
+	maxConnectionDuration := retry.AddJitter(disconnectTimeout, disconnectJitterMax)
+	logger.Info(("Websocket connection established."), logger.Fields{
+		"URL":                    cs.URL,
+		"ConnectTime":            startTime.Format(dateTimeFormat),
+		"ExpectedDisconnectTime": startTime.Add(disconnectTimeout).Format(dateTimeFormat),
+	})
+
+	timer := time.AfterFunc(maxConnectionDuration, func() {
+		err := cs.CloseClient(startTime, maxConnectionDuration)
+		cs.MetricsFactory.New(metricName).Done(err)
+		if err != nil {
+			logger.Warn(fmt.Sprintf("Attempted disconnecting; client already closed. %s", err))
```

Comment:
This seems like a duplicate log statement of the one on L626. Can we remove this and rely on the L626 log statement instead?